web/js/codeworld-mode.js

```diff
@@ -52,21 +52,55 @@ CodeMirror.defineMode('codeworld', (_config, modeConfig) => {
     const RE_INCOMMENT = /(?:[^{-]|-(?=$|[^}])|\{(?=$|[^-]))*/;
     const RE_ENDCOMMENT = /-}/;
 
+    const RE_OF = /of/;
+    const RE_LET = /let/;
+    const RE_IN = /in/;
+    const RE_WHERE = /where/;
+    const RE_DO = /do/;
+    const RE_CASE = /case/;
+
+    // Next non-comment or non-whitespace token
+    // is start of layout context
+    const TRIGGERED = 0;
+    // Parsing is not layout sensetive
+    const NORMAL = 1;
+
     function opening(bracket) {
         if (bracket === ')') return '(';
         if (bracket === ']') return '[';
         if (bracket === '}') return '{';
         return bracket;
     }
 
+    // Start new layout context.
+    function checkContextStarter(stream, state) {
+        if (stream.match(RE_LET) ||
+            stream.match(RE_WHERE) ||
+            stream.match(RE_DO) ||
+            stream.match(RE_CASE) ||
+            stream.match(RE_OF)
+        ) {
+            const nextChar = stream.peek();
+            // Check if it word like 'lettering' or 'offer'
+            if (nextChar === null || nextChar === undefined || /\s/.exec(nextChar)) {
```

Comment:
This looking at the next character seems ugly to me.  Would it work if you added `\b` to the end of the regular expressions above, instead?

Reply:
Yes, that's much better.